OpenRA.Platforms.Default/Sdl2GraphicsDevice.cs

```diff
@@ -189,7 +189,7 @@ public IHardwareCursor CreateHardwareCursor(string name, Size size, byte[] data,
 			}
 			catch (Exception ex)
 			{
-				throw new InvalidDataException("Failed to create hardware cursor `{0}` - {1}".F(name, ex.Message), ex);
+				throw new SDL2HardwareCursorException("Failed to create hardware cursor `{0}` - {1}".F(name, ex.Message), ex);
```

Comment:
What does a custom exception type win us here?

Reply:
Naming Accuracy: Failures I've observed here I've seen were not caused by invalid inputs to SDL.

Most instances of InvalidDataException I've seen in the codebase are pertinent to issues of incorrect data formats, where this one does not appear to be actionable by consumer in that sense, with SDL not rejecting the *format* of data it is supplied in callstacks I have been seeing, and the outer loop consuming it successfully retries on repeated attempt, in the current impl(has smell of a race condition).

I prefer to be able to keep my debugger catching first chance exceptions for anomalous paths, while omitting noisy Exception types which are not indicative of local implementation bugs. Where possible, I don't like overlooking exceptions which have a significant probability of indicating actionable malfunctions.

The cursor failure is a recurrent, but inconsistent repro on my windows environment, for which an InvalidDataException does not seem(to me) to be an accurate reflection of the problem from SDL. Based on threads from other consumers, I'm tempted to follow up on root cause fix with latest debug build of underlying SDL implementation as a separate issue, but that's the context.

Reply:
I'm ok with dropping the specific exception types here, if that's a concern on this PR.

Reply:
No, that sounds reasonable.  I haven't had time to take a closer look at this PR yet.